src/fides/api/models/sql_models.py

```diff
@@ -475,7 +475,9 @@ class PrivacyDeclaration(Base):
 
     features = Column(ARRAY(String), server_default="{}", nullable=False)
     legal_basis_for_processing = Column(String)
-    flexible_legal_basis_for_processing = Column(BOOLEAN())
+    flexible_legal_basis_for_processing = Column(
+        BOOLEAN(), server_default="t", nullable=False
```

Comment:
@adamsachs wanted to get your thoughts here before proceeding - as-is this maybe would have plus changes and compass changes somewhere.

An alternative, so plus/compass doesn't have to change, we still allow this to be nullable, but still make the default=True change.  That was my main objective here, so systems created outside of compass via the API or CLI would be True by default.  And then if compass returns as null, null/false basically mean the same thing.

Reply:
@pattisdr i think i'm good with the approach you're proposing. the compass changes would be pretty minimal to adjust for this -- we'd simply need to have the webesrvice assume that a null/empty value for this field in airtable should be returned as `False` in the API. i could even get those changes staged very quickly on a branch for us to test out end to end. after thinking about this a bit, i think it may actually make things a bit clearer for compass to model things this way - Airtable doesn't have an explicit `False` state, so there's really no way to distinguish `False`/`null`, so by not allowing `null` across our stack, we can basically _force_ `null` to signify `False` in airtable/compass.

if i'm understanding the alternative you're presenting correctly - i.e. keep it nullable but default to true - my gut tells me it will just lead to more confusion down the road. we'd basically be establishing different result states depending on whether we save the attribute to the DB with an explicit `None` vs. not specifying the attribute on a given save, right? in some ways that _could_ enable some more flexibility, but i think it comes at the expense of predictability/clear behavior down the road. we'd need to constantly keep track of whether the field is specified with `None`, unspecified, defaulted to a particular value, etc. through multiple layers of our stack - and i've always found that difficult. tl;dr is that i think just eliminating the nullable state (as you're proposing here) forces us to explicitly specify true/false across our stack, and i think the slight loss in flexibility is worth the clarity gained...

where do you foresee changes in plus? not doubting, just i don't know off the top of my head and i want to make sure i'm understanding potential impact accurately

Reply:
Great, just checking, I didn't know what it would take in compass. Let's go with what I have in this PR: I prefer that approach too.

Maybe plus doesn't need anything, maybe it's just compass thinking about it more.